include/zephyr/net/net_if.h

```diff
@@ -2643,16 +2643,47 @@ void net_if_unregister_link_cb(struct net_if_link_cb *link);
 void net_if_call_link_cb(struct net_if *iface, struct net_linkaddr *lladdr,
 			 int status);
 
+/* used to ensure encoding of checksum support in net_if.h and
```

Comment:
Doxygen comments missing, if these comments should be internal then add INTERNAL_HIDDEN guards

Reply:
I will add INTERNAL_HIDDEN guards; these definitions are only used to ensure the values of the enums in net_if.h and ethernet.h match.

Reply:
I have added doxygen comments for the actual enum values.